doc/rst/technotes/gpu.rst

```diff
@@ -0,0 +1,90 @@
+.. _readme-gpu:
+
+.. default-domain:: chpl
+
+=============================
+Support for GPU targeted code
+=============================
+
+Chapel includes preliminary work to target Nvidia GPUs using CUDA. This work is
+under active development and has not yet been tested under a wide variety of
+environments. We have tested it on systems with Nvidia Tesla P100 GPUs and CUDA
+11.0 and a system with Nvidia Tesla K20 GPUs with CUDA 10.2. The current
+implementation does not support multiple GPUs on one node and will only apply
+to certain `forall` loops.
+
+--------
+Overview
+--------
+
+To deploy code to a GPU embed the relevant code in an 'on' statement targeting
+the GPU locale (i.e. `here.getChild(1)`).
+
+Any arrays that are declared in the body of this `on` statement will be
+allocated on the GPU  Chapel will generate CUDA kernels for `forall` statements
+in the `on` block that are eligible.  Loops are eligible if they only make use
+of known compiler primitives that are fast and local are eligible for GPU code
+generation. Here "fast" means "safe to run in a signal handler" and "local"
+means "doesn't cause any network communication".
```

Comment:
I'd add something like "In practice, this means loops not containing any non-inlined function calls." to the end of this paragraph to address Engin's comment from the earlier issue.

Reply:
Fixed in 70f18f5727695ce8f553dd6920ac0714279336a3